wgsl/index.bs

```diff
@@ -689,84 +682,259 @@ An 8-bit byte is the most basic unit of [=host-shareable=] memory.
+structure member is explicitly annotated with [=size=] and / or [=align=]
+decorations, in which case those member decorations take precedence.
+
+The first structure member always has a zero byte offset from the start of the
+structure.
+
+Subsequent members have the following byte offset from the start of the structure:
+<p algorithm="structure member offset">
+  [=OffsetOf=](|S|, M<sub>N</sub>) = [=roundUp=]([=AlignOf=](|S|, M<sub>N</sub>), [=OffsetOf=](|S|, M<sub>N-1</sub>) + [=SizeOf=](|S|, M<sub>N-1</sub>)<br>
+  Where M<sub>N</sub> is the current member and M<sub>N-1</sub> is the previous member
+</p>
+
+Structure members must not overlap. Structure members decorated with the
+[=size=] attribute must have a size at least as large as the size of the
+member's type.
+
+The alignment of a structure is equal to the largest alignment of all of its
+members:
+<p algorithm="structure alignment">
+  [=AlignOf=](|S|) = max([=AlignOf=](|S|, M<sub>0</sub>), ... , [=AlignOf=](|S|, M<sub>N</sub>))
+</p>
+
+The size of a structure is equal to the offset plus the size of its last member,
+rounded to the next multiple of the structure's alignment:
+<p algorithm="structure size">
+  [=SizeOf=](|S|) = [=roundUp=]([=AlignOf=](|S|), [=OffsetOf=](|S|, |L|) + [=SizeOf=](|S|, |L|))<br>
+  Where |L| is the last member of the structure
+</p>
+
+<div class='example wgsl' heading='Layout of structures using implicit member strides and alignments'>
+  <xmp highlight='rust'>
+    [[block]] struct A {                           //             align(8)  size(24)
+        u : f32;                                   // offset(0)   align(4)  size(4)
+        v : f32;                                   // offset(4)   align(4)  size(4)
+        w : vec2<f32>;                             // offset(8)   align(8)  size(8)
+        x : f32;                                   // offset(16)  align(4)  size(4)
+        // -- implicit struct size padding --      // offset(20)            size(4)
+    };
+
+    [[block]] struct B {                           //             align(16) size(160)
+        a : vec2<f32>;                             // offset(0)   align(8)  size(8)
+        // -- implicit member alignment padding -- // offset(8)             size(8)
+        b : vec3<f32>;                             // offset(16)  align(16) size(12)
+        c : f32;                                   // offset(28)  align(4)  size(4)
+        d : f32;                                   // offset(32)  align(4)  size(4)
+        // -- implicit member alignment padding -- // offset(36)            size(12)
+        e : A;                                     // offset(40)  align(8)  size(24)
+        f : vec3<f32>;                             // offset(64)  align(16) size(12)
+        // -- implicit member alignment padding -- // offset(76)            size(4)
+        g : [[stride(24)]] array<A, 3>;            // offset(80)  align(8)  size(72)
```

Comment:
Is the stride intended to be present? Calling the example implicit make me think it should not be there.

Reply:
Removed